### ISSUE TITLE
[Option] Disable test menu

### ISSUE THREAD

**Issue by Aynshe** [jpd002/Play-#1298]
## Quick summary

Hi,

One to deactivate access to the test menu would be interesting to avoid a handling error.

THANKS ;)

**Comment by Keropon** [comment]
What kind of handling error? Test menu should be available at any time, as it's needed to config a lot of settings. I'm curious.

**Comment by Aynshe** [comment]
> What kind of handling error? Test menu should be available at any time, as it's needed to config a lot of settings. I'm curious.

it is not a question of completely disabling access to the menu, but of allowing an option to deactivate.  Once the calibration is done, the parameters are saved in a save file.  This prevents you from accessing the menu by mistake in-game.

**Comment by Keropon** [comment]
Then it would be completely unnecessary if the Test button could be mapped since you could just unmap it when you're done, right?

**Comment by Aynshe** [comment]
> Then it would be completely unnecessary if the Test button could be mapped since you could just unmap it when you're done, right?

This is also a solution.

**Comment by Hydreigon223** [comment]
Let's be realistic here. There are more reasons why one needs test mode access. If failing an important arcade run, entering and exiting test mode is quicker than rebooting the system. Operators critically need test mode access in case something is wrong with the system (bad controls or glitches). Your suggestion, in the previous case, would be the equivalent of if arcade machine distributors removed the physical test switch so only they could access it. It wasn't good practice and fortunately wasn't ever a trend.

I may have been a bit too elaborate on the test mode topic but let's get the elephant in the room here. I/O handling is screwed up beyond relief here because of attempting to hardwire the JVS logic to a PS2 controller. Unfortunately, I'd rather not go too much into detail here and take this to a new issue.

**Comment by jpd002** [comment]
Yeah, arcade inputs are not really well handled right now. I need to go back to the drawing board, it's much more involved that I initially imagined it. I'll consider your use case in my analysis.

**Comment by udance4ever** [comment]
hey I hope this Isn't too off topic as I can't find the answer documented anywhere:  What is the keyboard binding to enter "test mode" for an arcade title?

**Comment by jcstahl1** [comment]
> hey I hope this Isn't too off topic as I can't find the answer documented anywhere: What is the keyboard binding to enter "test mode" for an arcade title?

https://github.com/jpd002/Play-?tab=readme-ov-file#arcade-specific-controls
